grpc-messenger-service/src/main/java/com/kucuk/server/service/BlockService.java

```diff
@@ -0,0 +1,30 @@
+package com.kucuk.server.service;
+
+import com.kucuk.block.BlockServiceGrpc;
+import com.kucuk.block.BlockingCallRequest;
+import com.kucuk.block.BlockingCallResponse;
+import io.grpc.stub.StreamObserver;
+
+import java.time.Instant;
+
+public class BlockService extends BlockServiceGrpc.BlockServiceImplBase {
+
+    public BlockService() {
+    }
+
+    @Override
+    public void blockingCall(BlockingCallRequest request,
+                      StreamObserver<BlockingCallResponse> responseObserver) {
+        if (request.getBlockPeriodInMilliSeconds() > 0) {
+            try {
+                Thread.sleep(request.getBlockPeriodInMilliSeconds());
```

Comment:
we are not supposed to be blocking on the api right? https://stackoverflow.com/questions/57329994/how-can-i-manage-blocking-code-inside-grpc-java-server-method